### PR TITLE
Modernize modules in  Calibration/HcalAlCaRecoProducers

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Modernized the modules in this package which were referenced in #25090. This primarily meant converting from legacy modules to more thread-friendly versions.

#### PR validation:

The code compiles. The intent is for these changes to produce identical results.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: Calibration/HcalAlCaRecoProducers/src/AlCaGammaJetProducer.cc [hunk 1]
```diff
@@ -70,25 +66,41 @@ class AlCaGammaJetProducer : public edm::EDProducer {
...
+  auto labelTrigger_ = iConfig.getParameter<edm::InputTag>("TriggerResults");
+  auto labelPFCandidate_ = iConfig.getParameter<edm::InputTag>("particleFlowInput");
+  auto labelVertex_ = iConfig.getParameter<edm::InputTag>("VertexInput");
+  auto labelPFMET_ = iConfig.getParameter<edm::InputTag>("METInput");
+  auto labelGsfEle_ = iConfig.getParameter<edm::InputTag>("gsfeleInput");
+  auto labelRho_ = iConfig.getParameter<edm::InputTag>("rhoInput");
+  auto labelConv_ = iConfig.getParameter<edm::InputTag>("ConversionsInput");
+  auto labelBeamSpot_ = iConfig.getParameter<edm::InputTag>("BeamSpotInput");
+  auto labelLoosePhot_ = iConfig.getParameter<edm::InputTag>("PhoLoose");
+  auto labelTightPhot_ = iConfig.getParameter<edm::InputTag>("PhoTight");
```
Comment (pohsun) [inline Calibration/HcalAlCaRecoProducers/src/AlCaGammaJetProducer.cc:103]: I don't really get your consideration of creating new objects here. Could you offer some explanation?

Comment (Dr15Jones) [reply Calibration/HcalAlCaRecoProducers/src/AlCaGammaJetProducer.cc:103]: Thanks for the catch! This was a left over from an early attempt to see if I could do away with the InputTags as member data. I'll fix that.

**Comment by pohsun** [comment]
please test

**Comment by pohsun** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**